frontend/src/components/Searchbar/Searchbar.tsx

```diff
@@ -0,0 +1,191 @@
+import React, { useState } from "react";
+import { ReactComponent as SearchIcon } from "../../assets/icons/search.svg";
+import { ReactComponent as FilterIcon } from "../../assets/icons/filter.svg";
+
+import SearchBarModal from "../Modal/SearchBarModal";
+import SearchBarModalContent from "./SearchBarModalContent";
+import CustomButton from "./CustomButton";
+import "./Searchbar.modules.scss";
+
+const presentationList = [
+  "react presentation",
+  "JS presentation",
+  "SASS presentation",
+  "Design presentation",
+];
+
+function Searchbar() {
+  const [showFilter, setShowFilter] = useState(false);
+  const [showModal, setShowModal] = useState(false);
+  const [searchValue, setSearchValue] = useState("");
+  const [type, setType] = useState("presentation");
+  const [category, setCategory] = useState("back-end");
+  const [searchList, setSearchList] = useState<string[]>([]);
```

Comment:
Not sure if it's better, but you can use an object as the state and then just change the key/value that you want to update.

For example:
```
const [searchbarState, setSearchbarState] = useState({
  showFilter: false,
  showModal: false,
});

const someHandler = () => {
  setSearchbarState({ ...searchbarState, showModal: true });
};
  ```